subgraphs/aave-forks/protocols/aave-v3/src/mapping.ts

```diff
@@ -0,0 +1,832 @@
+    );
+    return;
+  }
+  _handleCollateralConfigurationChanged(
+    marketId,
+    event.params.liquidationBonus,
+    event.params.liquidationThreshold,
+    event.params.ltv,
+    protocolData
+  );
+}
+
+export function handleReserveActive(event: ReserveActive): void {
+  const marketId = getMarketIdFromToken(event.params.asset);
+  if (!marketId) {
+    log.error(
+      "[handleCollateralConfigurationChanged]Failed to find market for asset {}",
+      [event.params.asset.toHexString()]
+    );
+    return;
+  }
+  _handleReserveActivated(marketId, protocolData);
+}
+
+export function handleReserveBorrowing(event: ReserveBorrowing): void {
+  const marketId = getMarketIdFromToken(event.params.asset);
+  if (!marketId) {
+    log.error("[handleReserveBorrowing]Failed to find market for asset {}", [
+      event.params.asset.toHexString(),
+    ]);
+    return;
+  }
+  if (event.params.enabled) {
+    _handleBorrowingEnabledOnReserve(marketId, protocolData);
+  } else {
+    _handleBorrowingDisabledOnReserve(marketId, protocolData);
+  }
+}
+
+export function handleReserveFrozen(event: ReserveFrozen): void {
+  const marketId = getMarketIdFromToken(event.params.asset);
+  if (!marketId) {
+    log.error("[handleReserveFrozen]Failed to find market for asset {}", [
+      event.params.asset.toHexString(),
+    ]);
+    return;
+  }
+  _handleReserveDeactivated(marketId, protocolData);
+}
+
+export function handleReservePaused(event: ReservePaused): void {
+  const marketId = getMarketIdFromToken(event.params.asset);
+  if (!marketId) {
+    log.error("[handleReservePaused]Failed to find market for asset {}", [
+      event.params.asset.toHexString(),
+    ]);
+    return;
+  }
+  _handleReserveDeactivated(marketId, protocolData);
+}
+
+export function handleReserveFactorChanged(event: ReserveFactorChanged): void {
+  const marketId = getMarketIdFromToken(event.params.asset);
+  if (!marketId) {
+    log.error(
+      "[handleReserveFactorChanged]Failed to find market for asset {}",
+      [event.params.asset.toHexString()]
+    );
+    return;
+  }
+  _handleReserveFactorChanged(
+    marketId,
+    event.params.newReserveFactor,
+    protocolData
+  );
+}
+
+/////////////////////////////////
+///// Lending Pool Handlers /////
+/////////////////////////////////
+export function handleReserveDataUpdated(event: ReserveDataUpdated): void {
+  const protocol = getOrCreateLendingProtocol(protocolData);
+  const marketId = getMarketIdFromToken(event.params.reserve);
+  if (!marketId || !Market.load(marketId.toHexString())) {
+    log.warning("[handleReserveDataUpdated]Market not found for reserve {}", [
+      event.params.reserve.toHexString(),
+    ]);
+    return;
+  }
+
+  const market = Market.load(marketId.toHexString())!;
+  const assetPriceUSD = getAssetPriceInUSDC(
+    Address.fromString(market.inputToken),
+    Address.fromString(protocol.priceOracle),
+    event.block.number
+  );
+
+  // TODO: better to pass market and protocol
```

Comment:
Why?

Reply:
I had the impression that passing loaded entities to functions is slightly more efficient than passing the ids and loading those same entities inside the functions - probably from David's subgraph coffee chat: https://discord.com/channels/953684103012683796/953686506990608435/1048306930105454613